itou/geo/tests/tests_utils.py

```diff
@@ -0,0 +1,65 @@
+from django.test import TestCase
+
+from itou.cities.factories import create_city_balaguier_dolt, create_city_petite_ile
+from itou.geo.factories import QPVFactory, ZRRFactory
+from itou.geo.utils import coords_to_geometry, geographic_eligibility_criteria
+from itou.users.factories import JobSeekerFactory
+
+
+class GeographicEligibilityCriteriaTest(TestCase):
+    def setUp(self) -> None:
+        QPVFactory(code="QP093028")
+        # in ZRR
+        create_city_balaguier_dolt()
+        ZRRFactory(insee_code="12018")
+        # Partially in ZRR
+        ZRRFactory(insee_code="97405")
+        create_city_petite_ile()
+
+    def test_no_eligibility_criterion(self):
+        # "Lambda" job seeker : no qpv | zrr
+        criteria = geographic_eligibility_criteria(JobSeekerFactory())
+        self.assertEqual({}, criteria)
+
+    def test_criteria_with_qpv(self):
+        user_in_qpv = JobSeekerFactory(
+            address_line_1="Rue du turfu",
+            post_code="93300",
+            city="Aubervilliers",
+            coords=coords_to_geometry("48.917735", "2.387311"),
+            geocoding_score=0.99,
+        )
+        criteria = geographic_eligibility_criteria(user_in_qpv)
+
+        self.assertEqual(criteria["address_in_qpv"], "Rue du turfu, 93300 Aubervilliers")
+
+    def test_criteria_with_zrr(self):
+        user_in_zrr = JobSeekerFactory(
+            address_line_1="Rue paumée",
+            post_code="12260",
+            city="Balaguier d'Olt",
+            geocoding_score=0.99,
+        )
+        criteria = geographic_eligibility_criteria(user_in_zrr)
+
+        self.assertEqual(criteria, {"commune_in_zrr": "Balaguier d'Olt"})
+
+    def test_criteria_with_partial_zrr(self):
+        user_partially_in_zrr = JobSeekerFactory(
+            address_line_1="Rue exotique",
+            post_code="97429",
+            city="Petite-Île",
+            geocoding_score=0.99,
+        )
+        criteria = geographic_eligibility_criteria(user_partially_in_zrr)
+
+        self.assertEqual(criteria, {"commune_partially_in_zrr": "Petite-Île"})
+
+    def test_warnings(self):
+        geographic_eligibility_criteria(JobSeekerFactory())
+        with self.assertLogs("itou.geo.utils", level="WARNING") as cm:
+            geographic_eligibility_criteria(JobSeekerFactory())
+
+            self.assertEqual(2, len(cm.output))
+            self.assertIn("Can't get geolocation of user", cm.output[0])
+            self.assertIn("Can't match INSEE code for user", cm.output[1])
```

Comment:
nit: essayer d'utiliser `assert` maintenant qu'on est en pytest, c'est BIEN plus puissant sur le diff des erreurs. 

`assert x == y` ou `assert x in y`

Reply:
Juste la force de l'inertie...